app/src/parser/parser.ts

```diff
@@ -146,9 +148,14 @@ export async function parse(file: string, srcUrl: string | null = null): Promise
     rawTracks = parseKml(file);
   }
   if (rawTracks.length) {
+    let p: Promise<any>[] = [];
     for (let i = 0; i < rawTracks.length; i++) {
-      await addGroundAltitude(rawTracks[i].fixes);
+      const me = Symbol();
```

Comment:
I think blocking here is ok-ish.

If you upload multiple tracks they would usually be located at ~ the same place and hopefully only the first one would have to download the data. The subsequent ones would re-use the cached data.

Using a queue in https://github.com/vicb/flyxc/blob/master/app/src/parser/parser.ts#L210-L212 looks more important to me. The code would probably need to be changed to have 2 different loops.

The first loop should download the required track in parallel (using a queue) and the second loop would do the lookup once the data is available.